### ISSUE TITLE
Possible memory leak in src/curl_stream.c git_curl_stream_new()

### ISSUE THREAD

**Issue by bubaflub** [libgit2/libgit2#3571]
Reading through the code I think I found a potential memory leak along an error path.  Relevant code snippet:
```
int git_curl_stream_new(git_stream **out, const char *host, const char *port)
{
	curl_stream *st;
	CURL *handle;
	int iport = 0, error;

	st = git__calloc(1, sizeof(curl_stream));
	GITERR_CHECK_ALLOC(st);

	handle = curl_easy_init();
	if (handle == NULL) {
		giterr_set(GITERR_NET, "failed to create curl handle");
		return -1;
	}

	if ((error = git__strtol32(&iport, port, NULL, 10)) < 0)
		return error;
	...
```

We calloc() `st` but along either error path listed above we don't appear to be freeing that memory.

**Comment by carlosmn** [comment]
Yes, in those cases we'll leak the memory. If you'd like to send a PR, we'll be glad to review it.

**Comment by carlosmn** [comment]
Fixed via #3578
